### ISSUE TITLE
Adaptive Expressions Float division

### ISSUE THREAD

**Issue by stefangomez** [microsoft/botbuilder-js#3753]
**Is your feature request related to a problem? Please describe.**
While microsoft/botbuilder-js#3527 is about fixing explicit floats, it doesn't address wanting to get a float as a return object from a division operation.

Consider the following expression: `div(11, 2)` the result would be `5`. What if we wanted `5.5`? As of the currently, I think it'd be impossible to achieve. `div(11.0, 2)` and even `div(float("11.0"), float(2))` won't work, even with the microsoft/botbuilder-js#3527 fix. Even then, if those two situations were accounted for somehow, it wouldn't fix when we wanted to explicitly get a precise number from a division operation.

**Describe the solution you'd like**
A new prebuilt function I believe is the only solution. Something like `divPrecise(11, 2, 2)` where the last argument would be the number of decimal places to return. I believe the return object would have to be a `string` to accurately return the correct value. That function should return `5.50` 

**Describe alternatives you've considered**
I think extending the existing `div` function could work, but I don't think it would be backward compatible given the current tests that exist, e.g. `div(5, 2, 2)` where each argument is an additional division operation.

**Additional context**
Just that I am really liking this library, and hope that I can use it for this case. I'm happy to take a stab at this function, but since it's a decent amount of work, would like to have the core team validate that it's not already possible using other methods I may have overlooked.

**Comment by joshgummersall** [comment]
@chrimc62, @Danieladu makes a good point about difficulties inherent to the language. We would have to detect this at a higher level in the processing pipeline before strings are converted to numbers (see [StackOverflow suggestion for detecting 11.0](https://stackoverflow.com/questions/7422072/javascript-how-to-detect-number-as-a-decimal-including-1-0)). I can imagine this is a fairly large change. Are any of the approaches @garypretty suggested suitable?

**Comment by mrivera-ms** [comment]
Reviewed by the SDK team. Closing as Not Needed.

**Comment by MatthiasGwiozda** [comment]
I find that this is a major bug, which should be fixed.

We are not able to precisely divide two numbers (!)

I have searched 3 hours for a possibility to divide two integers and get a precise result with a comma.
I cannot believe that this is actually not possible with adaptive-expressions.

There is no possibility...
- ... [to create custom functions with adaptivecards-templating](https://github.com/microsoft/AdaptiveCards/issues/6133)
- ... to divide precisely with standard functions

For the SDK team it may be Not Needed.
But for the users of this package it is very much needed.

**Comment by Danieladu** [comment]
hi @MatthiasGwiozda  sorry for the capability that the library could not provide the function what you need.
As a workaround, you could inject your custom functions into the library by yourself, such like this:
```
Expression.functions.add('contoso.sqrt', (args) => {
            let result = null;
            if (args[0] !== undefined) {
                const inputFloat = parseFloat(args[0]);
                result = Math.sqrt(inputFloat);
            }

            return result;
        });
```

the above code would inject a user's sqrt function with function name "contoso.sqrt", and the usage is simple, 
`contoso.sqrt(xx)`

**Comment by MatthiasGwiozda** [comment]
Hi @Danieladu. Thanks for your fast response.

custom Functions are not an options for us.
we are using adaptivecards-templating. And adaptivecards-templating is statically importing only the standard functions from adaptive-expressions.
This problem is described here: https://github.com/microsoft/AdaptiveCards/issues/6133

This means for us that we have to programm our own "adaptivecards-templating" logic just to implement the divPrecise function.
This is a lot more time consuming than it should be.

**Comment by Danieladu** [comment]
You could use the workaround if it is necessary. 
For example if you want to keep 3 decimal places
```
float(concat(string(div(a, b), '.', mod(div(a*1000, b), 1000))))
```

**Comment by MatthiasGwiozda** [comment]
> You could use the workaround if it is necessary. For example if you want to keep 3 decimal places
> 
> ```
> float(concat(string(div(a, b), '.', string(mod(div(a*1000, b), 1000)))))
> ```

@Danieladu thank you for the workaround.
Ive searched for a possibility like this.

This formula is so cumbersome, that there is a mistake with a bracket.
This is the corrected formula, which is actually working.
You can check it in the [adaptivecard designer](https://adaptivecards.io/designer/).

```JavaScript
float(concat(string(div(a, b)),'.',string(mod(div(a*1000, b), 1000))))
```

When this is currently the only possibility to divide precisely two numbers with adaptivecards-templating, it should be documented.
The next person should not invest so much time, like me, to find a practicable way to achieve this.

**Comment by Danieladu** [comment]
@MatthiasGwiozda  Thanks. I have raised a PR in doc to show the difference and provide the workaround.

**Comment by Diegorro98** [comment]
Hi!
I've used the provided workaround, but it doesn't work pretty well with divisions where the first decimal digits are zero.
For example:
`1/11` should be `0.0909` (with 4 precision digits), but with the given workaround, the result is `0.909`

But I found a way to solve it:
```
float(concat(string(div(a, b)),'.', join(select(range(1, decimal_places), exp_n, mod((div(div(mul(a, exp(10, decimal_places)), b), exp(10,(sub(decimal_places, exp_n))))), 10)), '')))
```
You will need to define `decimal_places` and set it to the number of decimal places you want.
